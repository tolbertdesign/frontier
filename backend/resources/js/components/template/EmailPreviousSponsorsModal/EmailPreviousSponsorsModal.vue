<template>
  <div class="m-4 rounded-xl overflow-hidden previous-sponsor-modal">
    <div class="card bg-blue">
      <div
        v-if="isSending"
        class="card-content pt-12 pb-12 px-4 text-white text-center"
      >
        <SendEnvelope />
        <p class="font-bold text-xl">
          {{ lang.email_send_to }}:
        </p>
        <p class="font-semibold pt-4">
          {{ previousSponsors.length }} <span>{{ lang.contact }}</span>
        </p>
      </div>
      <div
        v-else
        class="card-content pt-16 px-4 text-white text-center"
      >
        <SendEnvelope />
        <p class="font-medium text-lg">
          {{ lang.about_to_email_previous }}
        </p>
      </div>
      <div
        v-if="!isSending"
        class="card-footer border-t-0"
      >
        <div class="card-footer-item">
          <button
            class="w-24 sm:w-32 button inline-block mb-4 mx-2 is-danger is-rounded shadow font-bold"
            @click="close"
          >
            {{ lang.cancel }}
          </button>
          <button
            class="w-24 sm:w-32 button inline-block mb-4 mx-2 is-secondary is-rounded is-inverted shadow font-bold"
            @click="sendEmail"
          >
            {{ lang.send }}
          </button>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import Blur from '@/mixins/Blur'

export default {
  name: 'EmailPreviousSponsorsModal',
  mixins: [Blur],
  props: {
    previousSponsors: {
      type: Array,
      default: null,
    },
    participantUserId: {
      type: Number,
      default: null,
    },
  },
  data () {
    return {
      isSending: false,
    }
  },
  computed: {
    lang () {
      return this.$store.state.lang
    },
  },
  methods: {
    sendEmail () {
      const sponsors = this.previousSponsors

      if (sponsors.length === 0) {
        this.close()
        return
      }

      const sponsorIds = sponsors.map(sponsor => sponsor.id)

      axios.post('/v3/api/previous-contact-enroll', {
        participantUserId: this.participantUserId,
        sponsorIds: sponsorIds,
      }).then(response => {
        this.isSending = true
        this.$emit('previousSponsorsEmailed', response.data.emailed, response.data.skipped)

        setTimeout(() => {
          this.close()
        }, 3000)
      }).catch(error => {
        console.error(error)
        setTimeout(() => {
          this.close()
        }, 3000)
      })
    },
    close () {
      this.$emit('close')
      this.unBlur()
    },
  },
}
</script>
