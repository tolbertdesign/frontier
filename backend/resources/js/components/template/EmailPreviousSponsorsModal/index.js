import EmailPreviousSponsorsModal from './EmailPreviousSponsorsModal.vue'

export default EmailPreviousSponsorsModal
