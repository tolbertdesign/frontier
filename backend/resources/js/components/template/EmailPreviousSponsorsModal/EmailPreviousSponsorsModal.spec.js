import { shallowMount, createLocalVue } from '@vue/test-utils'
import EmailPreviousSponsorsModal from './EmailPreviousSponsorsModal.vue'
import SendEnvelope from '@/components/element/SendEnvelope.vue'

const localVue = createLocalVue()

const previousSponsors = [
  { id: 1 },
  { id: 2 },
]
const $store = {
  state: {
    lang: {
      email_send_to: 'email send to',
      contact: 'contact',
      about_to_email_previous: 'about to email previous',
      cancel: 'cancel',
      send: 'send',
    },
  },
}

describe('EmailPreviousSponsorsModal', () => {
  it('Snapshop test for EmailPreviousSponsorsModal', () => {
    const wrapper = shallowMount(EmailPreviousSponsorsModal, {
      localVue,
      components: {
        SendEnvelope,
      },
      mocks: {
        $store,
      },
      propsData: {
        previousSponsors: previousSponsors,
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
