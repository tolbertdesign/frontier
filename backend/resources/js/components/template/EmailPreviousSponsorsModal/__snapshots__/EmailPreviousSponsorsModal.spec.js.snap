// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailPreviousSponsorsModal Snapshop test for EmailPreviousSponsorsModal 1`] = `
<div class="m-4 rounded-xl overflow-hidden previous-sponsor-modal">
  <div class="card bg-blue">
    <div class="card-content pt-16 px-4 text-white text-center">
      <sendenvelope-stub></sendenvelope-stub>
      <p class="font-medium text-lg">about to email previous</p>
    </div>
    <div class="card-footer border-t-0">
      <div class="card-footer-item"><button class="w-24 sm:w-32 button inline-block mb-4 mx-2 is-danger is-rounded shadow font-bold">
          cancel
        </button> <button class="w-24 sm:w-32 button inline-block mb-4 mx-2 is-secondary is-rounded is-inverted shadow font-bold">
          send
        </button></div>
    </div>
  </div>
</div>
`;
